engine/gamesys/src/gamesys/scripts/script_label.cpp

```diff
@@ -210,19 +211,23 @@ static int SetText(lua_State* L)
 
     dmGameObject::HInstance instance = CheckGoInstance(L);
 
-    size_t len = 0;
-    const char* text = luaL_checklstring(L, 2, &len);
+    size_t text_len = 0;
+    const char* text = luaL_checklstring(L, 2, &text_len);
     if (!text)
     {
         return DM_LUA_ERROR("Expected string as second argument");
     }
 
-    lua_newtable(L);
-    lua_pushlstring(L, text, len);
-    lua_setfield(L, -2, "text");
+    uint32_t data_size = sizeof(dmGameSystemDDF::SetText) + text_len + 1;
+    if (data_size > dmMessage::DM_MESSAGE_MAX_DATA_SIZE)
+    {
+        return DM_LUA_ERROR("The label string is too long!");
+    }
+    uint8_t data[data_size];
 
-    uint8_t data[dmMessage::DM_MESSAGE_MAX_DATA_SIZE];
-    uint32_t data_size = dmScript::CheckDDF(L, dmGameSystemDDF::SetText::m_DDFDescriptor, (char*)data, sizeof(data), -1);
+    dmGameSystemDDF::SetText* message = (dmGameSystemDDF::SetText*)data;
+    message->m_Text = (const char*)sizeof(dmGameSystemDDF::SetText);
+    dmStrlCpy((char*)data + sizeof(dmGameSystemDDF::SetText), text, text_len + 1);
```

Comment:
An alternative is to use memcpy() and then setting the final byte to 0.
Benefits: less code to run and fewer code dependencies.

Reply:
I believe setting the final byte to 0 is not needed since the C string is already null terminated?
```
memcpy((void*)(data + sizeof(dmGameSystemDDF::SetText)), text, text_len + 1);
```

Reply:
Correct, if you copy the null termination, then that works too.